json-cache/service-worker.js

```diff
@@ -0,0 +1,53 @@
+// [Working example](/serviceworker-cookbook/json-cache/).
+
+var CACHE_NAME = 'dependencies-cache';
+
+self.addEventListener('install', function(event) {
+  // Perform the install step:
+  // - Load a JSON file from server
+  // - Parse as JSON
+  // - Add files to the cache list
+
+  event.waitUntil(
+    caches.open(CACHE_NAME)
+      .then(function(cache) {
+        // With the cache opened, load a JSON file containing an array of files to be cached
+        return fetch('files-to-cache.json').then(function(response) {
+          // Once the contents are loaded, convert the raw text to JSON
```

Comment:
To be precise, I would reword this as `convert the raw text to a JavaScript object` as JSON **is** actually raw text.